debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java

```diff
@@ -148,23 +149,28 @@ protected void delaySnapshotIfNeeded(ChangeEventSourceContext context) throws In
     /**
      * Prepares the taking of a snapshot and returns an initial {@link SnapshotContext}.
      */
-    protected abstract SnapshotContext<O> prepare(ChangeEventSourceContext changeEventSourceContext) throws Exception;
+    protected abstract SnapshotContext<P, O> prepare(P partition) throws Exception;
```

Comment:
The `ChangeEventSourceContext` argument wasn't used by any of the implementations (including the incubating ones). Unlike the `offset` property which isn't initialized in the `SnapshotContext` constructor and set somewhere down the line, we can and want to initialize `partition` right away, hence the change here and in constructors of `SnapshottingTask` subclasses.

Reply:
Ok, we can do that. I think we had the context for all the snapshot lifecycle methods, but if it's used nowhere for this method, no need to have it here indeed.